File: core/baker-annotations/src/main/java/com/ing/baker/recipe/annotations/ProcessId.java
package com.ing.baker.recipe.annotations;

import javax.inject.Qualifier;
import java.lang.annotation.*;

/**
 * An annotation to be added to an argument of an action indicating that the process identifier should be injected
 * there.
 */
@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.PARAMETER)
public @interface ProcessId {
}

File: core/baker-annotations/src/main/kotlin/com/ing/baker/recipe/annotations/ProcessId.kt
package com.ing.baker.recipe.annotations

import javax.inject.Qualifier

/**
 * An annotation to be added to an argument of an action indicating that the process identifier should be injected
 * there.
 */
@Qualifier
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.VALUE_PARAMETER)
annotation class ProcessId
